### PR TITLE
MARKET-1304 Multiple Bar Chart

### PR DESCRIPTION
**Pull request by Saharij**
182399472-ae112d0d-d4d0-48b1-bc41-d5e8195787f6
182399487-3a7e6965-c77c-4b1a-ab28-c6c3636f65e9
Screenshot 2022-08-03 at 20 43 22

### CODE REVIEW

File: components/bl-multiple-bar-chart-component/src/decorate-number.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  for (const number of restNumbers) {
+    if (number !== '0') {
+      lastNumbers.push(number);
+    }
+  }
+  
+  return lastNumbers.length ? `${numbers}.${lastNumbers.reverse().join('')}` : numbers;
+};
+
+export const decorateNumber = number => {
```
Comment (v-excelsior) [inline components/bl-multiple-bar-chart-component/src/decorate-number.js:15]: https://github.com/Backendless/ui-builder-library/pull/27#discussion_r938798505
https://github.com/Backendless/ui-builder-library/pull/27#discussion_r938800690

File: components/bl-multiple-bar-chart-component/src/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+
+import { validate } from './validate';
+import { Bar } from './bar';
+
+const { cn } = BackendlessUI.CSSUtils;
+const percentageMarksList = [0, 25, 50, 75, 100];
```
Comment (v-excelsior) [inline components/bl-multiple-bar-chart-component/src/index.js:7]: Could we add property for that? Also, should be a variant without marks(if we not provide them)

File: components/bl-multiple-bar-chart-component/src/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        <div className="chart__items">
+          <div className="chart__grids">
+            { percentageMarksList.map(item => (
+              <div className="chart__y-grid-line" style={{ left: `${item}%`}} />
+            )) }
+          </div>
+          { chartData.length > 0 && chartData.map(({ objectId, name, goal, progress }) => (
+            <Bar
+              key={ objectId }
+              name={ name ? name : 'Name' }
```
Comment (v-excelsior) [inline components/bl-multiple-bar-chart-component/src/index.js:34]: `name || 'Name'` ?

File: components/bl-multiple-bar-chart-component/src/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+              goal={ goal ? goal : 1 }
+              progress={ progress ? progress : 0 }
```
Comment (v-excelsior) [inline components/bl-multiple-bar-chart-component/src/index.js:36]: `progress || 0` ?

File: components/bl-multiple-bar-chart-component/src/validate.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    return items.map(item => ({
+      ...item,
+      objectId: getId(),
+    }));
+  }
+    
+  return items;
+};
+
+const getId = () => {
```
Comment (v-excelsior) [inline components/bl-multiple-bar-chart-component/src/validate.js:16]: @Valodya I assume it's good candidate to share between all components.

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-multiple-bar-chart-component/src/validate.js (hunk 5 above)
Comment (Valodya) [reply components/bl-multiple-bar-chart-component/src/validate.js:16]: `BackendlessUI.UUID.short()`

**Pull request merged**